sink/kafka.go

```diff
@@ -0,0 +1,114 @@
+package sink
+
+import (
+	"fmt"
+	"github.com/Shopify/sarama"
+	log "github.com/Sirupsen/logrus"
+	"os"
+	"strings"
+	"time"
+)
+
+// KafkaSink ...
+type KafkaSink struct {
+	// Kafka brokers to send metrics to
+	Brokers []string
+	// Kafka topic
+	Topic string
+
+	producer sarama.SyncProducer
+
+	stopCh chan interface{}
+	putCh  chan []byte
+}
+
+// NewKafka ...
+func NewKafka() (*KafkaSink, error) {
+
+	brokers := os.Getenv("SINK_KAFKA_BROKERS")
+	if brokers == "" {
+		return nil, fmt.Errorf("[sink/kafka] Missing SINK_KAFKA_BROKERS")
+	}
+
+	brokerList := strings.Split(brokers, ",")
+	log.Debugf("[sink/kafka] Kafka brokers: %s", strings.Join(brokerList, ", "))
+
+	topic := os.Getenv("SINK_KAFKA_TOPIC")
+	if topic == "" {
+		return nil, fmt.Errorf("[sink/kafka] Missing SINK_KAFKA_TOPIC")
+	}
+	log.Debugf("[sink/kafka] Kafka topic: %s", topic)
+
+	config := sarama.NewConfig()
+	config.Producer.Return.Successes = true
+
+	producer, err := sarama.NewSyncProducer(brokerList, config)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
just return `nil, err`

Reply:
but i didn't wanted if it's can't initialize producer silently, i wanted it to die!